src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -1,145 +1,101 @@
 import okhttp3.HttpUrl;
-import org.apache.commons.lang.StringUtils;
 import org.openqa.selenium.WebDriverException;
 
-public class UrlBuilder {
+public class UrlBuilder extends BaseUrlBuilder {
 
 
-  protected String env;
-  protected EnvType envType;
+  private final String wikiName;
   private Boolean forceHttps;
   private Boolean forceLanguageInPath;
-  protected static final String HTTP_PREFIX = "http://";
+  private String language;
 
-  public UrlBuilder() {
-    this.env = Configuration.getEnv();
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = Configuration.getForceHttps();
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
-  }
-
-  public UrlBuilder(String env, Boolean forceHttps) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
-    this.forceHttps = forceHttps;
-    this.forceLanguageInPath = Configuration.getForceLanguageInPath();
-  }
-
-  public UrlBuilder(String env, Boolean forceHttps, Boolean forceLanguageInPath) {
-    this.env = env;
-    this.envType = Configuration.getEnvType(this.env);
+  private UrlBuilder(String wiki, String env, Boolean forceHttps, Boolean forceLanguageInPath, String language) {
+    super(env);
+    this.wikiName = wiki;
     this.forceHttps = forceHttps;
     this.forceLanguageInPath = forceLanguageInPath;
+    this.language = language;
   }
 
-  public String getLanguageForWiki() {
-    return Configuration.getWikiLanguage() == null ? Configuration.DEFAULT_LANGUAGE : Configuration.getWikiLanguage();
+  public static UrlBuilder createUrlBuilder() {
+    return createUrlBuilderForWikiAndLang(getWikiName(), getWikiLanguage());
   }
 
-  public String normalizePageName(String pageName) {
-    return pageName.replace(" ", "_");
+  public static UrlBuilder createUrlBuilderForWiki(String wiki) {
+    return createUrlBuilderForWikiAndLang(wiki, getWikiLanguage());
   }
 
-  public String getUrlForPageWithWWW(String pageName) {
-    return getUrlForWiki(Configuration.getWikiName(), true) + pageName;
+  public static UrlBuilder createUrlBuilderForWikiAndLang(String wiki, String language) {
+    return new UrlBuilder(wiki, getEnv(), getForceHttps(), getForceLanguageInPath(), language);
   }
 
-  public String getUrlForPage(String pageName) {
-    return getUrlForWiki() + pageName;
-  }
-
-  public String getUrlForPage(String wikiName, String pageName) {
-    return getUrlForWiki(wikiName) + pageName;
-  }
-
-  public String getUrlForPage(Page page) {
-    if (page.getWikiPath() == null) {
-      return getUrlForWiki(page.getWikiName(), page.getWikiLanguage(), false);
-    }
-    return getUrlForPath(page.getWikiName(), page.getWikiLanguage(), page.getWikiPath());
-  }
-
-  public String getUrlForPage(Page page, String qs) {
-    return appendQueryStringToURL(page.getUrl(), qs);
+  public String normalizePageName(String pageName) {
+    if ("".equals(pageName))
+      throw new WebDriverException("Page name is missing");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "Page name is missing" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)